tests/recovery.test.ts

```diff
@@ -0,0 +1,164 @@
+import { WorkflowContext, OperonWorkflow, OperonTestingRuntime } from "../src/";
+import { generateOperonTestConfig, setupOperonTestDb } from "./helpers";
+import { OperonConfig } from "../src/operon";
+import { OperonTestingRuntimeImpl, createInternalTestRuntime } from "../src/testing/testing_runtime";
+import { OperonWorkflowRecoveryUrl } from "../src/httpServer/server";
+import request from "supertest";
+
+describe("recovery-tests", () => {
+  let config: OperonConfig;
+  let testRuntime: OperonTestingRuntime;
+
+  beforeAll(async () => {
+    config = generateOperonTestConfig();
+    await setupOperonTestDb(config);
+  });
+
+  beforeEach(async () => {
+    testRuntime = await createInternalTestRuntime([LocalRecovery, ExecutorRecovery], config);
+  });
+
+  afterEach(async () => {
+    await testRuntime.destroy();
+  });
+
+  /**
+   * Test for the default local workflow recovery.
+   */
+  class LocalRecovery {
+    static resolve1: () => void;
+    static promise1 = new Promise<void>((resolve) => {
+      LocalRecovery.resolve1 = resolve;
+    });
+
+    static resolve2: () => void;
+    static promise2 = new Promise<void>((resolve) => {
+      LocalRecovery.resolve2 = resolve;
+    });
+
+    static cnt = 0;
+
+    @OperonWorkflow()
+    static async testRecoveryWorkflow(ctxt: WorkflowContext, input: number) {
+      if (ctxt.authenticatedUser === "test_recovery_user" && ctxt.request.url === "test-recovery-url") {
+        LocalRecovery.cnt += input;
+      }
+
+      // Signal the recovery is done.
```

Comment:
I understand you can only enter this branch if the  `invoke` workflow ran and increment `cnt`

Reply:
I can only enter this branch if the workflow is executed more than once.